ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -513,13 +524,32 @@ def process_cluster_cli_params(config):
     skip_download_client = get_cli_param(config, "skip_download_client")
     if skip_download_client:
         ocsci_config.DEPLOYMENT["skip_download_client"] = True
+    re_trigger_failed_tests = get_cli_param(config, "--re-trigger-failed-tests")
+    if re_trigger_failed_tests:
+        ocsci_config.RUN["re_trigger_failed_tests"] = os.path.expanduser(
+            re_trigger_failed_tests
+        )
 
 
 def pytest_collection_modifyitems(session, config, items):
     """
     Add Polarion ID property to test cases that are marked with one.
     """
-    for item in items:
+
+    re_trigger_failed_tests = ocsci_config.RUN.get("re_trigger_failed_tests")
+    if re_trigger_failed_tests:
+        junit_report = JUnitXml.fromfile(re_trigger_failed_tests)
+        cases_to_re_trigger = []
+        for suite in junit_report:
+            cases_to_re_trigger += [_case.name for _case in suite if _case.result]
+    for item in items[:]:
```

Comment:
Why do we need a new copy of the items list and not just removing it from the original items? And also where is this new copy of the items list will be stored if it is not assigned to a variable like `duplicate_list = items[:]`

Reply:
@jimdevops19 this is one of the bad think you can do. You shouldn't delete the item from the list you are iterating over as it can leads to weird situations like you will miss some of the item and so on..  e.g. this article:  https://thispointer.com/python-remove-elements-from-a-list-while-iterating/  so this is the safe way how to iterate over all values and do not change iterator .  You can find many more articles on this problematic , this is one of the first I just googled.